### ISSUE TITLE
Unhelpful errors

### ISSUE THREAD

**Issue by mscharley** [cjoudrey/graphql-schema-linter#53]
Output:

```
$ graphql-schema-linter $(bin/find-schemas graphql)

It looks like you may have hit a bug in graphql-schema-linter.

It would be super helpful if you could report this here: https://github.com/cjoudrey/graphql-schema-linter/issues/new

TypeError: Cannot read property 'length' of null
    at /usr/src/app/node_modules/graphql-schema-linter/lib/source_map.js:30:57
    at Array.reduce (native)
    at SourceMap._computeOffsets (/usr/src/app/node_modules/graphql-schema-linter/lib/source_map.js:28:20)
    at new SourceMap (/usr/src/app/node_modules/graphql-schema-linter/lib/source_map.js:16:25)
    at Configuration.getSchema (/usr/src/app/node_modules/graphql-schema-linter/lib/configuration.js:92:26)
    at run (/usr/src/app/node_modules/graphql-schema-linter/lib/runner.js:43:30)
    at Object.<anonymous> (/usr/src/app/node_modules/graphql-schema-linter/lib/cli.js:15:32)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
```

I'm not sure what exactly leads to this, this is while I'm trying to clean up after a massive brainstorming session with the rest of my team. When I dig up more information I'll report back with what I find.

**Comment by cjoudrey** [comment]
Thanks for the report @mscharley.

I was just looking at the line that caused the error, could it be that you have a `.graphql` file that has no line breaks?

https://github.com/cjoudrey/graphql-schema-linter/blob/e1800eb82cb048135832f344fe7d188487c29e1e/src/source_map.js#L14

In that case the `.match` would return `null` and it would 💥.

Sounds like a legit issue that should be fixed.

**Comment by mscharley** [comment]
Ah yes, that's it. I had an empty file which is what triggered it. Deleting that since the contents had been moved elsewhere solved this issue.

**Comment by mscharley** [comment]
In a similar vein, the following sorts of errors are showing up too. The raw errors are useful, but the CLI reports them as potential bugs:

```
It looks like you may have hit a bug in graphql-schema-linter.

It would be super helpful if you could report this here: https://github.com/cjoudrey/graphql-schema-linter/issues/new

Error: Type "PersonLikeConnection" not found in document.
    at typeDefNamed (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/utilities/buildASTSchema.js:270:13)
    at produceType (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/utilities/buildASTSchema.js:240:19)
    at produceOutputType (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/utilities/buildASTSchema.js:249:46)
```

```
It looks like you may have hit a bug in graphql-schema-linter.

It would be super helpful if you could report this here: https://github.com/cjoudrey/graphql-schema-linter/issues/new

Error: Query fields must be an object with field names as keys or a function which returns suc
h an object.
    at invariant (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/jsutils/invariant.js:19:11)
    at defineFieldMap (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/type/definition.js:354:54)
```

**Comment by cjoudrey** [comment]
💯 agree with you on catching these errors and reporting them instead of bailing as we do atm.

I'll open a PR to fix both of these problems.

**Comment by cjoudrey** [comment]
Will be fixed in next release by #163.
